gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/FsBenchmark.java

```diff
@@ -195,28 +247,48 @@ private void benchmarkRead(
 
                 byte[] readBuffer = new byte[readSize];
+                Path file_to_io = testFile;
+
+                if (!benchmarkRead) {
+                  Random r = new Random();
+                  r.nextBytes(readBuffer);
+                  String random_file = String.format("/test-%s-%03d.bin", random_id, file_counter);
+                  file_to_io = new Path(testFile.toString() + random_file);
+                }
 
                 initLatch.countDown();
                 startLatch.await();
                 try {
                   for (int j = 0; j < numReads; j++) {
-                    try (FSDataInputStream input = fs.open(testFile)) {
-                      long readStart = System.nanoTime();
-                      long fileBytesRead = 0;
-                      int bytesRead;
-                      do {
-                        long readCallStart = System.nanoTime();
-                        bytesRead = input.read(readBuffer);
-                        if (bytesRead > 0) {
-                          fileBytesRead += bytesRead;
-                          readCallBytes.accept(bytesRead);
-                        }
-                        readCallTimeNs.accept(System.nanoTime() - readCallStart);
-                      } while (bytesRead >= 0);
-
-                      readFileBytes.accept(fileBytesRead);
-                      readFileTimeNs.accept(System.nanoTime() - readStart);
+                    long readStart = System.nanoTime();
+                    long fileBytesRead = 0;
+                    if (benchmarkRead) {
+                      try (FSDataInputStream input = fs.open(file_to_io)) {
+                        int bytesRead;
+                        do {
+                          long readCallStart = System.nanoTime();
+                          bytesRead = input.read(readBuffer);
+                          if (bytesRead > 0) {
+                            fileBytesRead += bytesRead;
+                            readCallBytes.accept(bytesRead);
+                          }
+                          readCallTimeNs.accept(System.nanoTime() - readCallStart);
+                        } while (bytesRead >= 0);
+                      }
+                    } else {
+                      try (FSDataOutputStream output = fs.create(file_to_io)) {
+                        int bytesRead;
+                        do {
+                          long readCallStart = System.nanoTime();
+                          output.write(readBuffer);
+                          fileBytesRead += readSize;
+                          readCallBytes.accept(readSize);
+                          readCallTimeNs.accept(System.nanoTime() - readCallStart);
+                        } while (fileBytesRead < totalSize);
+                      }
                     }
+                    readFileBytes.accept(fileBytesRead);
+                    readFileTimeNs.accept(System.nanoTime() - readStart);
```

Comment:
Please simplify this logic is appropriate methods

Reply:
Quite a few local variables accessed and updated here. Separating it out in a function will be tricky.